prereise/gather/griddata/hifld/data_process/profiles.py

```diff
@@ -100,3 +104,95 @@ def build_wind(wind_plants, download_directory, year):
         directory=download_directory,
     )
     return profiles
+
+
+def filter_anomalies_impute(series, capacity, rel_min=-1, rel_max=2, max_count=100):
+    """Given a time-series of generation and an assumed capacity, identify anomalous
+    points and replace them by linear imputation (unless there are too many, in which
+    case return a copy of the original.
+
+    :param pandas.Series/numpy.ndarray series: 1-D data to filter and impute (MW).
+    :param int/float capacity: assumed capacity (MW).
+    :param int/float rel_min: lower screening value (relative to ``capacity``).
+    :param int/float rel_max: upper screening value (relative to ``capacity``).
+    :param int max_count: the maximum number of anomalies that can be replaced. If more
+        than this number exist, the series will be returned as is.
+    :return: (*pandas.Series*) -- data, imputed as applicable.
+    :raise ValueError: if series isn't a 1-D object
+    """
+    if len(series.shape) != 1:
+        raise ValueError("series must be of one-dimentional shape")
+    suspected_anomalies = (series > rel_max * capacity) | (series < rel_min * capacity)
+    output = series.copy()
+    if suspected_anomalies.sum() > max_count or suspected_anomalies.sum() == 0:
+        return output
+    output.loc[suspected_anomalies] = float("nan")
```

Comment:
Should we give a warning message if there are too many anomalies?

Reply:
That would help the user

Reply:
Done. An example of the new warning: `Too many suspected anomalies for PSEI, none will be filtered`.

The reason for this edge-case logic is for when there is a large disparity between the maximum capacity from Form 860 and the magnitude of the profiles. In this case, there will be many times where the generation profile is too high, and we don't want to erase these values: we want to use them later to normalize the profile, so that we get the proper shape, not a clipped shape.

Reply:
Makes sense